apple/swift-docc
# add the ability to set a subset of feature flags from Info.plist

Bug/issue #, if applicable: rdar://126305435

## Summary

This PR adds a new field to the Info.plist to set feature flags without using the command line. Right now, this only adds support for the overloads feature, but this mechanism is in place to facilitate adopting experimental features that affect how a bundle is written.

## Dependencies

None

## Testing

To test this feature, make a copy of the `OverloadedSymbols.docc` test bundle and modify its Info.plist to look like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleName</key>
	<string>ShapeKit</string>
	<key>CFBundleDisplayName</key>
	<string>ShapeKit</string>
	<key>CFBundleIdentifier</key>
	<string>com.shapes.ShapeKit</string>
	<key>CFBundleVersion</key>
	<string>0.1.0</string>
	<key>CDExperimentalFeatureFlags</key>
	<dict>
		<key>ExperimentalOverloadedSymbolPresentation</key>
		<true/>
	</dict>
</dict>
</plist>
```

Steps:
1. Overwrite `Tests/SwiftDocCTests/Test Bundles/OverloadedSymbols.docc/Info.plist` with the above content, or copy `OverloadedSymbols.docc` into a new location and overwrite its Info.plist there.
2. Ensure that `DOCC_HTML_DIR` is set to a recent main branch of Swift-DocC-Render or `swift-docc-render-artifact`.
3. `swift run docc preview Tests/SwiftDocCTests/Test\ Bundles/OverloadedSymbols.docc` (replace the bundle path if you copied it to a new location)
4. Navigate to `OverloadedProtocol` and ensure that only one `fourthTestMemberName(test:)` is curated in the topic groups.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
